circuit/src/layered/circuit.rs

```diff
@@ -361,14 +384,20 @@ impl<C: GKRFieldConfig> Circuit<C> {
         self.rnd_coefs_identified = true;
     }
 
-    pub fn fill_rnd_coefs<T: Transcript<C::ChallengeField>>(&mut self, transcript: &mut T) {
+    pub fn fill_rnd_coefs<T: Transcript<C::ChallengeField>>(
+        &mut self,
+        transcript: &mut T,
+        mpi_config: &MPIConfig,
+    ) {
```

Comment:
I suppose this method will also be used in verifier, that is ran by only the root process...